src/components/utils.ts

```diff
@@ -74,6 +74,9 @@ export const getAlertsAndRules = (
   // Add `rule` object to each alert
   const alerts = _.flatMap(rules, (rule) => rule.alerts.map((a) => ({ rule, ...a })));
 
+  // `alert.rule.labels` contains external labels which we want to include with `alert.labels`.
+  alerts.map((alert) => (alert.labels = { ...alert.rule.labels, ...alert.labels }));
```

Comment:
```suggestion
  // Add external labels to all `rules[].alerts[].labels`
  rules.map(...);
  
  const alerts = _.flatMap(rules, (rule) => rule.alerts.map((a) => ({ rule, ...a })));
```

As mentioned in my other comment, I think we need to add the external labels not just to all the `alerts[].labels`, but also to all the `rules[].alerts[].labels`. So if we add them to `rules[].alerts[].labels` first, the existing `const alerts = _.flatMap(rules, (rule) => rule.alerts.map((a) => ({ rule, ...a })))` line should take care of copying them to `alerts[].labels`.

Reply:
Thanks, good catch! I've updated this in the latest push.